.github/utils/check_imports.py

```diff
@@ -2,64 +2,74 @@
     """
     imported = []
     failed = []
+    exclude_subdirs = (exclude_subdirs or []) + ["__pycache__"]
 
-    sys.path.insert(0, directory)
+    # Add the root directory to the Python path
+    sys.path.insert(0, root_dir)
+    base_path = Path(root_dir)
 
-    for root, _, files in os.walk(directory):
-        # skip directories without __init__.py
-        if not '__init__.py' in files:
+    for root, _, files in os.walk(root_dir):
+        if any(subdir in root for subdir in exclude_subdirs):
             continue
 
-        init_path = os.path.join(root, '__init__.py')
-
-        # skip haystack/__init__.py to avoid circular imports
-        if init_path.endswith(f'{directory}/__init__.py'):
-            continue
-
-        # convert filesystem path to Python module name
-        relative_path = os.path.relpath(root, directory)
-        module_name = relative_path.replace(os.path.sep, '.')
-        if module_name == '.':
-            module_name = os.path.basename(directory)
-
-        try:
-            importlib.import_module(module_name)
-            imported.append(module_name)
-        except:
-            failed.append({
-                'module': module_name,
-                'traceback': traceback.format_exc()
+        # Convert path to module format
+        module_path = ".".join(Path(root).relative_to(base_path.parent).parts)
+        python_files = [f for f in files if f.endswith('.py')]
+
+        # Try importing package and individual files
+        for file in python_files:
+            try:
+                if file == "__init__.py":
+                    module_to_import = module_path
+                else:
+                    module_name = os.path.splitext(file)[0]
+                    module_to_import = f"{module_path}.{module_name}" if module_path else module_name
+
+                importlib.import_module(module_to_import)
+                imported.append(module_to_import)
+            except:
+                failed.append({
+                    'module': module_to_import,
+                    'traceback': traceback.format_exc()
                 })
 
     return imported, failed
 
-
 def main():
     """
-    This script checks that all Haystack packages can be imported successfully.
-    This can detect several issues.
-    One example is forgetting to use a forward reference for a type hint coming
-    from a lazy import.
+    This script checks that all Haystack modules can be imported successfully.
+    This includes both packages and individual Python files.
+    This can detect several issues, such as:
+    - Syntax errors in Python files
+    - Missing dependencies
+    - Circular imports
+    - Incorrect type hints without forward references
     """
-    print("Checking imports from Haystack packages...")
-    imported, failed = validate_package_imports(directory="haystack")
+    # Add any subdirectories you want to skip during import checks
+    exclude_subdirs = ["__pycache__", "testing"]
```

Comment:
`__pycache__` is excluded in the implementation of `validate_module_imports` so adding it here is not wrong but redundant.

Reply:
Ok. I removed it from the list and explained in the comment above.